### PR TITLE
Added function to get heart rate from processed hr (avg pixel value per frame) json file

### PR DESCRIPTION
**Pull request by itismeghasyam**
the function getHrFromJson takes in a processed HR json file and outputs a list, which has measured hr and confidence of the measurement for each color channel. The window is 10s and each consecutive window overlaps 50% with the previous one. So the hr readings will be from times 0-10s, 5-15s, 10-20s, .... and so on

### CODE REVIEW

**Comment by philerooski** [comment]
You need to export your function in the namespace so that it can be called https://github.com/itismeghasyam/mpowertools/blob/master/NAMESPACE

File: R/getHrFromJson.R [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+getHrFromJson <- function(hrJsonFileLoc, windowLen = 10){
+
+  #############################################################
+  # Required Sub Functions
+  #############################################################
+
+  # Bandpass and sorted mean filter the given signal
+  
+  getfilteredsignal <- function(x, mforder = 33, bpforder = 128, freqRange=c(2,25), samplingRate){
```
Comment (th1vairam) [inline R/getHrFromJson.R]: Consider moving sub-functions outside the main function.

File: R/getHrFromJson.R [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    for (i in seq((mforder+1)/2, length(x)-(mforder-1)/2,1)){
+      tempseq <- sort(x[(i-((mforder-1)/2)):((i+((mforder-1)/2)))])
+      y[i] = x[i]-mean(tempseq[1:(length(tempseq)-1)])
+    }
+    return(y)
+  }
+  
+  # Given a processed time series find its period using autocorrelation and then convert it to HR (bpm)
+  
+  getHR <- function(x, samplingRate, minHR = 40, maxHR=200){
```
Comment (th1vairam) [inline R/getHrFromJson.R]: Move the sub-function outside the main function

File: R/getHrFromJson.R [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    
+    return(c(hr, confidence))
+  }
+  
+  #############################################################
+  # Main Code Block
+  #############################################################
+  
+  # If no json file exists
+  if(is.na(hrJsonFileLoc)){return('No JSON file') }
```
Comment (th1vairam) [inline R/getHrFromJson.R]: Try to keep the output format constant. Like the final variable dat with a tag called error with value NA and return the list. Check one of the tremor code for sample.

File: R/getHrFromJson.R [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+                 error = function(e){ NA })
+  if(is.na(dat)){return('JSON file read error') }
+  
+  # Get sampling rate
+  samplingRate = length(dat$timestamp)/(dat$timestamp[length(dat$timestamp)] - dat$timestamp[1])
+  
+  # Convert window length from seconds to samples
+  windowLen = round(samplingRate*windowLen)
+  
+  # Apply pre processing filter signal between freqRange
```
Comment (th1vairam) [inline R/getHrFromJson.R]: Try to make these as inputs to the function. So that one can change in the future.

**Comment by itismeghasyam** [comment]
Just moved the sub functions outside the getHrFromJson function (y)

File: R/getHrFromJson.R (hunk 4 above)
Comment (itismeghasyam) [reply R/getHrFromJson.R]: These parameters are set based on normal physiological changes (freq range and mforder (for a max of 220BPM)). I can make bpforder a input to the function, but then again anything lower on a 2-scale(i,e 64,32,16...) was performing poorly, should I still make the changes ?

Comment (itismeghasyam) [reply R/getHrFromJson.R]: Made changes to include freqrange and bpforder as inputs to the function. I left mpforder as is, as it is a calculated quantity based on how fast the human heart can beat (we are assuming 220BPM here)

**Pull request merged**